pkg/helpers/newapp/app/imageref_test.go

```diff
@@ -89,6 +89,83 @@ func TestBuildConfigOutput(t *testing.T) {
 	}
 }
 
+func TestImageRefAsImageStreamSuggestedNameContainsSlashes(t *testing.T) {
+	objectName := "some/test/name-1"
+	host := "test-host.org"
+	tag := "latest"
+	namespace := "namespace-name"
+
+	expectedObjectNameNoSlashes := "some-test-name-1"
+
+	tests := []struct {
+		imageRef              *ImageRef
+		expectedSuggestedName string
+	}{
+		{
+			imageRef: &ImageRef{
+				ObjectName:    objectName,
+				AsImageStream: false,
+			},
+			expectedSuggestedName: objectName,
+		},
+
+		{
+			imageRef: &ImageRef{
+				ObjectName:    objectName,
+				AsImageStream: true,
+			},
+			expectedSuggestedName: expectedObjectNameNoSlashes,
+		},
+
+		{
+			imageRef: &ImageRef{
+				Reference: reference.DockerImageReference{
+					Registry:  host,
+					Tag:       tag,
+					Namespace: namespace,
+					Name:      objectName,
+				},
+				AsImageStream: false,
+			},
+			expectedSuggestedName: objectName,
+		},
+
+		{
+			imageRef: &ImageRef{
+				Reference: reference.DockerImageReference{
+					Registry:  host,
+					Tag:       tag,
+					Namespace: namespace,
+					Name:      objectName,
+				},
+				AsImageStream: true,
+			},
+			expectedSuggestedName: expectedObjectNameNoSlashes,
+		},
+
+		{
+			imageRef: &ImageRef{
+				Stream: &imagev1.ImageStream{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      objectName,
+						Namespace: namespace,
+					},
+				},
+				AsImageStream: true,
+			},
+			expectedSuggestedName: expectedObjectNameNoSlashes,
+		},
+	}
+
+	for _, test := range tests {
+		suggestedName, _ := test.imageRef.SuggestName()
+
+		if suggestedName != test.expectedSuggestedName {
+			t.Errorf("Wrong suggested name, expected '%s', actual '%s'", test.expectedSuggestedName, suggestedName)
```

Comment:
Since all of your tests are using the same `expectedSuggestedName` it is going to be very difficult for someone to determine which test case is failing. I would rather see the test format used where each case has a `name` which is output when the test runs if it fails: Here is an example: https://gobyexample.com/testing

Reply:
It takes some branching to figure out type of `ImageRef` test is running against, so I just added test names.